src/tms_service.ts

```diff
@@ -110,6 +131,108 @@ export class TMSService extends AbstractEmsService {
   }
 
+  /*
+  This static function transforms a style to use the passed language
+  */
+  public static transformLanguage(
+    style: EmsVectorStyle,
+    lang: keyof typeof TMSService.SupportedLanguages
+  ): EmsVectorStyle {
+    const omtLang = TMSService.SupportedLanguages[lang]?.omtCode;
+
+    if (!omtLang) {
+      throw new Error(`Language [${lang}] is not supported`);
+    }
+
+    style.layers
+      .filter((l) => l.layout && l.layout.hasOwnProperty('text-field'))
+      .forEach((l) => {
+        const { layout } = l as SymbolLayerSpecification;
+        if (layout) {
+          const label = layout['text-field'];
+
+          if (typeof label === 'string') {
+            // Capture {name:xx} labels
+            const labelMatch = label.match(/\{name:(.{2})\}/);
+            if (labelMatch && labelMatch[1] != omtLang) {
+              // Only apply if the languages are different
+              layout['text-field'] = [
+                'coalesce',
+                ['get', `name:${omtLang}`],
+                ['get', `name:${labelMatch[1]}`],
+              ];
+            } else {
+              if (label === '{name:latin}\n{name:nonlatin}') {
+                // Capture the common pattern {name:latin}\n{name:nonlatin}
+                layout['text-field'] = [
+                  'coalesce',
+                  ['get', `name:${omtLang}`],
+                  ['concat', ['get', 'name:latin'], '\n', ['get', 'name:nonlatin']],
+                ];
+              } else if (label.includes('name')) {
+                // Capture any other label using `name`
+                layout['text-field'] = [
+                  'coalesce',
+                  ['get', `name:${omtLang}`],
+                  ['get', 'name:latin'],
+                ];
+              }
+            }
+          }
+        }
+      });
+    return style;
+  }
+
+  /*
+
+  */
+  static computeLayer(layer: LayerSpecification, color: string) {
+    if (['background', 'fill', 'line', 'symbol'].indexOf(layer.type) !== -1 && layer.paint) {
+      const paint = layer.paint as layerPaintProperty;
+      const types = Object.keys(paint).filter((key) => {
+        return (
+          [
+            'background-color',
+            'circle-color',
+            'circle-stroke-color',
+            'fill-color',
+            'fill-extrusion-color',
+            'fill-outline-color',
+            'icon-color',
+            'icon-halo-color',
+            'line-color',
+            'text-color',
+            'text-halo-color',
+          ].indexOf(key) !== -1
+        );
+      }) as Array<keyof layerPaintProperty>;
+      const sources = types.map((type) => {
+        const paintColor = paint[type];
+        if (paintColor) {
+          //const colorDesaturated = desaturateColor(paintColor);
+          const colorHueChanged = colorizeColor(paintColor, color);
+          return { [type]: colorHueChanged };
+        }
+      });
+      layer.paint = Object.assign({}, paint, ...sources);
```

Comment:
nit: I didn't test this, so I could be wrong, but I think you don't need `Object.assign` if you do this:

```suggestion
      layer.paint = { ...sources, ...paint }
```

Reply:
We refactored the method so this code is not present anymore, and there are no other `Object.assign` calls in the class.